File: template/python/timestamp.py
import os
import json

def timestamp(file_path, event_id, start_time, end_time):
    duration_seconds = end_time - start_time

    event = {
        "name": event_id,
        "start": start_time,
        "end": end_time,
        "duration": duration_seconds
    }

    events = []
    if os.path.exists(file_path):
        try:
            with open(file_path, 'r') as file:
                content = file.read()
                events = json.loads(content)
        except Exception as e:
            print(f"Error reading the file: {e}")

    events.append(event)

    try:
        with open(file_path, 'w') as file:
            json.dump(events, file, indent=4)
    except Exception as e:
        print(f"Error writing to the file: {e}")

File: template/java/timestamp.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.time.Instant;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public static void timestamp(String filePath, String eventId, double startTime, double endTime) throws JSONException {
    double durationSeconds = endTime - startTime;

    JSONObject event = new JSONObject();
    event.put("name", eventId);
    event.put("start", startTime);
    event.put("end", endTime);
    event.put("duration", durationSeconds);

    JSONArray events = new JSONArray();
    File file = new File(filePath);
    if (file.exists()) {
        try {
            String content = new String(Files.readAllBytes(Paths.get(filePath)));
            events = new JSONArray(content);
        } catch (IOException e) {
            System.err.println("Error reading the file: " + e.getMessage());
        }
    }

    events.put(event);

    try (FileWriter fileWriter = new FileWriter(filePath)) {
        fileWriter.write(events.toString(4));
    } catch (IOException e) {
        System.err.println("Error writing to the file: " + e.getMessage());
    }
}